src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,128 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(last, value, null);
+        if (size == 0) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size && index >= 0) {
```

Comment:
you can avoid checking condition index >= 0 if you will move checkIndex to the top of the method